### PR TITLE
Updated to grails 2.3.7 and jaxrs 0.10

### PR DESCRIPTION
**Pull request by 573f**
Hi Angel,

I work with Sam Teasdale (who contacted you about a new version). This pull request represents the changes we needed to make to get it to compile and work with grails 2.3.7.

### CODE REVIEW

**Comment by aruizca** [comment]
Hi @573f !

Thanks for the PR

If you read the README.md of the project it says that in order to use Grails 2.3+ You should explicitly add the jaxrs 0.10 dependency in the BuildConfig.groovy file of your host project. This will override the jaxrs dependency included in the plugin.

Current version of the swagger4jaxrs plugin (2.0) works with any version of grails 2.x as long as you follow the installation instructions. If I merge your changes, it will only work for 2.3+ Grails versions.

Please try with current version of the plugin and let me know if you have any problems.

Cheers,
Angel.

**Comment by 573f** [comment]
Hi Angel,

Thanks for the quick reply. When we follow the installation instructions using grails 2.3.7, and include the following in BuildConfig.groovy:

``` groovy
plugins {
    ...
    compile ':jaxrs:0.10'
    compile ':swagger4jaxrs:0.2'
    ...
}
```

We get the following compilation errors:

```
Compilation error: startup failed:
/Users/skubik/FRG_DataDefinition/target/work/plugins/swagger4jaxrs-0.2/grails-app/controllers/com/nerderg/grails/plugins/swagger4jaxrs/ShowRestApiController.groovy: 3: unable to resolve class com.wordnik.swagger.jaxrs.config.BeanConfig
 @ line 3, column 1.
   import com.wordnik.swagger.jaxrs.config.BeanConfig
   ^

/Users/skubik/FRG_DataDefinition/target/work/plugins/swagger4jaxrs-0.2/grails-app/controllers/com/nerderg/grails/plugins/swagger4jaxrs/ShowRestApiController.groovy: 9: unable to resolve class com.wordnik.swagger.jaxrs.config.BeanConfig 
 @ line 9, column 5.
       BeanConfig swaggerConfig
       ^

2 errors
```

**Comment by 573f** [comment]
We have noticed that when we can get the 0.2 version to work, it has some issues when running in grails 2.3.7 and jaxrs-0.10:
- the basePath is not set properly (it's missing the application context)
- MetaClass properties are included in the JSON output and the jackson-annotation @JsonIgnoreProperties annotation is not honored.

We are finding that when we build the 0.3-SNAPSHOT version (current master revision) locally, it solves the first issue. The second issue appears to be solved in a later version of swagger (> 1.3.5 - see https://github.com/swagger-api/swagger-core/issues/519).

In version 0.2, both the swagger and jackson plugins are down-level. This is what motivated our initial question about a possible upcoming release.

Would you be willing to simply publish the 0.3-SNAPSHOT? That would solve some of our most pressing issues.

**Comment by 573f** [comment]
I've amended the pull request to include updated swagger and jackson plugins. This resolves both the basePath and MetaClass issues. I bumped the version to 0.3.1-SNAPSHOT to avoid conflicts with your current version.

File: Swagger4jaxrsGrailsPlugin.groovy [hunk 1]
```diff
@@ -5,7 +5,7 @@ import org.codehaus.groovy.grails.commons.GrailsApplication
 import com.wordnik.swagger.jaxrs.config.BeanConfig
 
 class Swagger4jaxrsGrailsPlugin {
-    def version = "0.3-SNAPSHOT"
+    def version = "0.3.1-SNAPSHOT"
```
Comment (aruizca) [inline Swagger4jaxrsGrailsPlugin.groovy:8]: Please leave it as 0.3

File: grails-app/conf/BuildConfig.groovy [hunk 2]
```diff
@@ -25,7 +25,7 @@ grails.project.dependency.resolution = {
     }
 
     plugins {
-        compile(':jaxrs:0.8')
+        compile(':jaxrs:0.10')
```
Comment (aruizca) [inline grails-app/conf/BuildConfig.groovy:28]: You should be able to override this dependency in your host app. It should be equivalent  IMO

**Pull request closed without merging**